### PR TITLE
Add error handling

### PR DESCRIPTION
**Pull request by dongbohu**
This PR checks `source` and `target` parameters in `querypair` endpoint.

### CODE REVIEW

File: dj_hetmech_app/views.py [hunk 1]
```diff
@@ -64,13 +63,23 @@ def polish_pathcounts(self, source_id, target_id, pathcounts_data):
...
-        target_id = request.query_params.get('target', None)
+        if source_id is None:
+            return Response({'error': 'source parameter not found in URL.'})
         try:
             source_node = Node.objects.get(pk=source_id)
+        except:
+            return Response({'error': 'source node not found in database.'})
+
+        # Validate "target" parameter
+        target_id = request.query_params.get('target', None)
```
Comment (dhimmel) [inline dj_hetmech_app/views.py:82]: Is `query_params` a dict? If so None is default value for get. Still okay to include if want to be explicit.

**Review by dhimmel — COMMENTED** [review]
We should also be returning the proper error codes right?

File: dj_hetmech_app/views.py (hunk 1 above)
Comment (dongbohu) [reply dj_hetmech_app/views.py:82]: Yes, I want to be explicit here.

**Review by dhimmel — APPROVED** [review]

**Comment by dhimmel** [comment]
@dongbohu feel free to squash merge if you're done.

**Pull request merged**